File: 28-01-23/insert_into_a_binary_search_tree/sol.py
#iterative

class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        req=TreeNode(val)
        if not root:
            return req
        cur=root
        parent=None
        while(cur!=None):
            parent=cur
            if(val<cur.val):
                cur=cur.left
            else:
                cur=cur.right
        if(val<parent.val):
            parent.left=req
        else:
            parent.right=req
        return root

File: 28-01-23/insert_into_a_binary_search_tree/sol.java
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        TreeNode req=new TreeNode(val);
        if(root==null){
            return req;
        }
        TreeNode cur=root;
        TreeNode parent=null;
        while(cur!=null){
            parent=cur;
            if(val<cur.val){
                cur=cur.left;
            }
            else{
                cur=cur.right;
            }
        }
        if(val<parent.val){
            parent.left=new TreeNode(val);
        }
        else{
            parent.right=new TreeNode(val);
        }
        return root;
        
    }
}
